apps/src/code-studio/utils.js

```diff
@@ -70,3 +70,7 @@ export function environmentSpecificCookieName(name) {
 
   return `${name}_${rack_env}`;
 }
+
+export function getRootDomainFromHostname(hostname) {
+  return hostname.split('.').slice(-2).join('.');
```

Comment:
"Root domain" doesn't seem to be a well-known term - in fact, [it may have a _different_ meaning when talking about DNS](https://en.wikipedia.org/wiki/Root_name_server#Root_domain).  Therefore, it'd be helpful to have a doc comment explaining what this method is doing.

Reply:
Second: Your PR description implies that this cookie uses `.code.org` as a domain but this helper gives `code.org` without the leading `.` and I don't see it added later, either.  Is this inferred by the cookie library or something?

Reply:
Hrm, I found quite a few references to root domain as being the domain name, but that's mainly in SEO and site-specific circles.  Is there a better name to use?  I'll document the method at any rate.

It's interesting, but by specifying the domain when setting the cookie, we get it to show up as a cookie named with the `.` in front, and that lets the cookie apply to all subdomains too.  https://stackoverflow.com/a/9618243

Reply:
:+1: I don't have a better term, just hunting around I didn't find anything.  TIL about the cookie domain!